src/System.Buffers/src/System/Buffers/BufferPool.cs

```diff
@@ -41,7 +36,8 @@ public T[] RentBuffer(int size, bool clearBuffer = false)
             }
             else
             {
-                buffer = _buckets[SelectBucketIndex(size)].Rent();
+                int index = SelectBucketIndex(size);
+                buffer = LazyInitializer.EnsureInitialized(ref _buckets[index], () => new BufferBucket<T>(GetMaxSizeForBucket(index), NUMBER_OF_BUFFERS_IN_BUCKET)).Rent();
```

Comment:
I know this is already a class and all, but if you use the delegate you will actually allocate for it. Could we get around with the extra allocation?